### DISCUSSION TITLE
pico isnt connecting to thonny

### DISCUSSION THREAD

**Discussion by UmutDemirel06** [micropython/micropython#10659] [RP2040 / Pico]
Hi ı am very new in this community ı am trying to code my pico using thonny but it  gives a weird error 
![1](https://user-images.githubusercontent.com/59196975/216766280-9c8141fc-c3a8-4231-8715-7c0f0d9b5c90.PNG)
![2](https://user-images.githubusercontent.com/59196975/216766282-e414a2a1-c9ef-487a-b5ab-408771938572.PNG)
![3](https://user-images.githubusercontent.com/59196975/216766283-be37ee5a-e8dc-4206-a1ce-fd1825e04d09.PNG)
![4](https://user-images.githubusercontent.com/59196975/216766284-87d04625-48a0-4ba7-8072-c9b95491dd36.PNG)
![5](https://user-images.githubusercontent.com/59196975/216766285-12bb863f-8b01-4d6f-9b22-f6c0406c26de.PNG)
 ![6](https://user-images.githubusercontent.com/59196975/216766518-e581210d-26dd-4955-a441-3d55434350bf.PNG)
as you can see ı tried to follow the most common way

**Comment by MariaA2005** [comment]
After uploading micropython, you need to disconnect the pico to take it out of bootloader mode, then connect it again.

**Reply by UmutDemirel06** [reply]
pico does that on its own but it still same

**Comment by beetlegigg** [comment]
Recently I put a new micropython on my picoW and, after disconnecting it and reconnecting it and trying to connect to it to Thonny (which was running on an rpi4) I got a similar situation.   Also I have found that instead of selecting 'Try to detect port automatically' I get a more reliable connection if I use the dropdown to select the port.   However, In my case,  selecting the port shown in the drop down still did not work.   I tried closing Thonny and executing it again, but still not luck, so I then tried the more brutal tactic  of  rebooting the rpi4.  When it was back up the Thonny drop down then showed a choice of 3 ports and by selecting the newly showing port I was back in business with my pico.    -  So a long spiel to suggest you try a reboot !

**Reply by UmutDemirel06** [reply]
ı would appreciate it if you could recommend a source for rebooting pico

**Comment by beetlegigg** [comment]
@Hopesforgypsy  - I was referring to rebooting the computer on which Thonny was running, in my case an rpi4.   As regards the pico, disconnecting it from a power source will do the trick.

**Reply by UmutDemirel06** [reply]
oh well i tried that but it didnt change anything

**Comment by beetlegigg** [comment]
I think, from looking at your original post, you were downloading mp to the pico via Thonny.   I suggest you download the latest mp directly from their web site, and copy it onto you pico.  (get your pico ready by holding the bootsel button down whilst applying the power)  When you've copied the latest to the pico, disconnect pico from power, reconnect it to your computer that runs Thonny, fire up Thonny and hopefully you should be in business.

**Reply by UmutDemirel06** [reply]
let me try that real quick

**Reply by UmutDemirel06** [reply]
![7](https://user-images.githubusercontent.com/59196975/216781901-affa6f05-1db9-4ef0-9099-4c562af8143b.PNG)

**Reply by beetlegigg** [reply]
No a success I see, but at least you now seem to be able to connect to your pico and try to run a program.   Can you see the files on your pico via the Thonny file view?   Make sure you dont have a boot.py or main.py that is auto running code on boot up.

**Reply by UmutDemirel06** [reply]
lemme see real quick

**Reply by UmutDemirel06** [reply]
![8](https://user-images.githubusercontent.com/59196975/216782302-6bd55968-c4c9-4470-9147-a1ab60a48fb3.PNG)
are you talking about this?

**Reply by cve2022** [reply]
> ![8](https://user-images.githubusercontent.com/59196975/216782302-6bd55968-c4c9-4470-9147-a1ab60a48fb3.PNG) are you talking about this?

Hello @Hopesforgypsy , this is what Windows shows when you press the boot button on the Pico and connect it to the USB port (or reset it for that matter - more on this below).

You drag and drop the micropython binary (with extension .uf2) to this folder, it is saved into Pico's flash and after that the Pico  is reset and starts Micropython. At this point Thonny should be able to connect to your Pico.

One suggestion I have is you try Putty to communicate with your Pico (find what is the COM interface with Windows device manager). You should have the REPL prompt (>>>).

Below is how I have all my Picos set:
![image](https://user-images.githubusercontent.com/108999978/216785406-b15d082e-6f48-4eef-96bc-cc53641c5006.png)
The button is between pin 30 (RUN) and 28 (GND), thus it resets the Pico when pressed. Very practical when one is developing to avoid disconnecting and reconnecting the board very often. Of course, I often use the breadboard to test circuits but still, even if it is not the casie, the breadboard provides physical support and protects the pins.

**Reply by UmutDemirel06** [reply]
well ı nuked pico then put rp2-pico-20220618-v1.19.1.uf2 file it disconnected on its own but some how I can't seem to connect pico to thonny
![1](https://user-images.githubusercontent.com/59196975/216786401-8149fe1f-f425-4b99-bba9-66df9f16ff43.PNG)

**Reply by UmutDemirel06** [reply]
thats the pico in device manager

**Reply by beetlegigg** [reply]
After loading the mp onto the pico, and the pico then auto disconnects, did you also remove the power to the pico, reconnect the power again, and only then try to connect to it with Thonny?   Thats the sequence I'm in the habit of doing, especially as my mac computer can get into a strop about the auto disconnect happening without giving it an 'eject' command and can necessitate a mac reboot.  It may well be different on a PC but it would be worth doing that sequence to see if you can get a connection.

**Reply by UmutDemirel06** [reply]
yeah every time i tried to program it ı made sure to unplug pico from usb but thonny just doesnt see the com

**Reply by beetlegigg** [reply]
Try making sure Thonny is not running when you connect your pico.  Then fire up Thonny and if you don't get a connection, check again the interpreter and port selection in Thonny.   Also, if you are connected thought a usb hub, try connecting direct to the PC usb port.

**Reply by beetlegigg** [reply]
Also, perhaps try a different usb cable.  Sometimes the cable can cause problems, or so I've read.

**Reply by UmutDemirel06** [reply]
> Also, perhaps try a different usb cable. Sometimes the cable can cause problems, or so I've read.

that might acctualy be case but also ı use same cable for arduino ide

**Reply by UmutDemirel06** [reply]
> Try making sure Thonny is not running when you connect your pico. Then fire up Thonny and if you don't get a connection, check again the interpreter and port selection in Thonny. Also, if you are connected thought a usb hub, try connecting direct to the PC usb port.

i tried everything here no luck

**Reply by cve2022** [reply]
> well ı nuked pico then put rp2-pico-20220618-v1.19.1.uf2 file it disconnected on its own but some how I can't seem to connect pico to thonny ![1](https://user-images.githubusercontent.com/59196975/216786401-8149fe1f-f425-4b99-bba9-66df9f16ff43.PNG)

Desculpe, mas este é um fórum em inglês, não acho que adianta postar uma mensagem em romeno (???) e esperar que um não romeno (???) entenda. :smile: 

Translating: I am sorry, but this is an English based forum, I don't think it is of any use posting a message in Romanian (???) and expect a non-Romanian(???) person to understands it. :smile:

If it were textual, I could try using google translate. I am not sure to which screen the above correspond comparing with my Windows 10 environment (COM ports in device manager screen):
![image](https://user-images.githubusercontent.com/108999978/216791646-93cc0d3b-7cae-496c-b729-6d09b117cdd4.png)

When I double click on COM8, I get the properties screen with some information on the device and the "working OK" status.

![image](https://user-images.githubusercontent.com/108999978/216791618-a365d9a7-114d-4377-8961-e1d8251698af.png)

I assume the same occurs with yours, right?

Have you tried connecting with putty?

**Reply by UmutDemirel06** [reply]
> > well ı nuked pico then put rp2-pico-20220618-v1.19.1.uf2 file it disconnected on its own but some how I can't seem to connect pico to thonny ![1](https://user-images.githubusercontent.com/59196975/216786401-8149fe1f-f425-4b99-bba9-66df9f16ff43.PNG)
> 
> Desculpe, mas este é um fórum em inglês, não acho que adianta postar uma mensagem em romeno (???) e esperar que um não romeno (???) entenda. 😄
> 
> Translating: I am sorry, but this is an English based forum, I don't think it is of any use posting a message in Romanian (???) and expect a non-Romanian(???) person to understands it. 😄
> 
> If it were textual, I could try using google translate. I am not sure to which screen the above correspond comparing with my Windows 10 environment (COM ports in device manager screen): ![image](https://user-images.githubusercontent.com/108999978/216791646-93cc0d3b-7cae-496c-b729-6d09b117cdd4.png)
> 
> When I double click on COM8, I get the properties screen with some information on the device and the "working OK" status.
> 
> ![image](https://user-images.githubusercontent.com/108999978/216791618-a365d9a7-114d-4377-8961-e1d8251698af.png)
> 
> I assume the same occurs with yours, right?
> 
> Have you tried connecting with putty?

oh sorry ı tought just port numbers would be useful ı will try to see com thing and what is putty?

**Reply by UmutDemirel06** [reply]
> > well ı nuked pico then put rp2-pico-20220618-v1.19.1.uf2 file it disconnected on its own but some how I can't seem to connect pico to thonny ![1](https://user-images.githubusercontent.com/59196975/216786401-8149fe1f-f425-4b99-bba9-66df9f16ff43.PNG)
> 
> Desculpe, mas este é um fórum em inglês, não acho que adianta postar uma mensagem em romeno (???) e esperar que um não romeno (???) entenda. 😄
> 
> Translating: I am sorry, but this is an English based forum, I don't think it is of any use posting a message in Romanian (???) and expect a non-Romanian(???) person to understands it. 😄
> 
> If it were textual, I could try using google translate. I am not sure to which screen the above correspond comparing with my Windows 10 environment (COM ports in device manager screen): ![image](https://user-images.githubusercontent.com/108999978/216791646-93cc0d3b-7cae-496c-b729-6d09b117cdd4.png)
> 
> When I double click on COM8, I get the properties screen with some information on the device and the "working OK" status.
> 
> ![image](https://user-images.githubusercontent.com/108999978/216791618-a365d9a7-114d-4377-8961-e1d8251698af.png)
> 
> I assume the same occurs with yours, right?
> 
> Have you tried connecting with putty?

![2](https://user-images.githubusercontent.com/59196975/216950229-d7e4b358-c6cf-4382-a01e-d7ae6bcc8a9a.PNG)
hey ı just looked into it and it says its working ok but it says from Bluetooth connection standard series com 5

**Reply by UmutDemirel06** [reply]
so problem comes from there probly

**Reply by cve2022** [reply]
> so problem comes from there probly

COM3 and COM4 are linked to BT. Your Pico should appear as COM5 or higher.
![image](https://user-images.githubusercontent.com/108999978/217038089-c1621628-868f-44ca-92e5-1cbe86b945e8.png)
The pico here appears as "USB Serial Device COM8".

Where did you get the binary you are trying to program your Pico with? 

Another idea, after you copy the binary to the Pico, if you disconnect and reconnect it to the USB, does the USB drive still appear on Windows Explore? Spoiler alert: it shouldn't.
![image](https://user-images.githubusercontent.com/108999978/217038802-f3a7b5d8-b26a-456b-bb57-eaae537255ac.png)

**Reply by UmutDemirel06** [reply]
binary ? are talking about a file named like this (rp2-pico-20220618-v1.19.1.uf2) if so it's from raspberry own site 
when ı copy the binary to pico it automatically disconnects and connects again but doesn't appear as USB device

**Reply by robert-hh** [reply]
A .uf2 file is a binary. If you say: it connects again, do you see the drive symbol again? If yes, then there may be something wrong with the firmware. Please try a firmware file from the MicroPython size https://micropython.org/download/rp2-pico/, e.g. this one: https://micropython.org/resources/firmware/rp2-pico-20230203-unstable-v1.19.1-852-g9ea64a36a.uf2

**Reply by UmutDemirel06** [reply]
ı don't see drive ıcon again it just makes usb disconnect and connect sound but i will still try the firmware

**Reply by UmutDemirel06** [reply]
i think the problem might be related to the computer i am using because pico doesn't appear as a com device

**Reply by UmutDemirel06** [reply]
> A .uf2 file is a binary. If you say: it connects again, do you see the drive symbol again? If yes, then there may be something wrong with the firmware. Please try a firmware file from the MicroPython size https://micropython.org/download/rp2-pico/, e.g. this one: https://micropython.org/resources/firmware/rp2-pico-20230203-unstable-v1.19.1-852-g9ea64a36a.uf2

i tried but didn't change anything

**Reply by robert-hh** [reply]
So the proper driver may be missing.

**Reply by UmutDemirel06** [reply]
might be lemme check real quick

**Reply by cve2022** [reply]
> i think the problem might be related to the computer i am using because pico doesn't appear as a com device

This is a wild guess, but... I think it is worth trying it.

Disconnect your Pico from the USB port and any USB-Serial device you have.

Have Device Manager show hidden devices
![image](https://user-images.githubusercontent.com/108999978/217048213-6ca9939d-0db6-42c2-bbd4-38e98f4b1a6c.png)

Delete all shaded devices that appear (COM5, COM6 etc.).
![image](https://user-images.githubusercontent.com/108999978/217048479-5f23eaa2-0b9d-4a8b-93f7-082bea612a71.png)

Do not check "Exclude driver for this device" for the time being. Connect your Pico. It should be recognized as COM5. If it does not, then try excluding the driver this time. If it does not work, then I have no idea what is happening. My next troubleshooting step would be booting this PC with a USB flash drive with a live Ubuntu distribution and try to see if your pico is recognized (as ttyUSB0 or, maybe ttyACM0).

**Reply by UmutDemirel06** [reply]
> > i think the problem might be related to the computer i am using because pico doesn't appear as a com device
> 
> This is a wild guess, but... I think it is worth trying it.
> 
> Disconnect your Pico from the USB port and any USB-Serial device you have.
> 
> Have Device Manager show hidden devices ![image](https://user-images.githubusercontent.com/108999978/217048213-6ca9939d-0db6-42c2-bbd4-38e98f4b1a6c.png)
> 
> Delete all shaded devices that appear (COM5, COM6 etc.). ![image](https://user-images.githubusercontent.com/108999978/217048479-5f23eaa2-0b9d-4a8b-93f7-082bea612a71.png)
> 
> Do not check "Exclude driver for this device" for the time being. Connect your Pico. It should be recognized as COM5. If it does not, then try excluding the driver this time. If it does not work, then I have no idea what is happening. My next troubleshooting step would be booting this PC with a USB flash drive with a live Ubuntu distribution and try to see if your pico is recognized (as ttyUSB0 or, maybe ttyACM0).

lemme try this too

**Reply by UmutDemirel06** [reply]
well it seems like i need to do that ubuntu thing can you explain a little more

**Reply by cve2022** [reply]
> well it seems like i need to do that ubuntu thing can you explain a little more

If you are not familiar with Linux, then it is not a good approach. My next one would be trying your Pico on a friend's PC and see with Device Manager if it is recognized.

**Reply by UmutDemirel06** [reply]
> > well it seems like i need to do that ubuntu thing can you explain a little more
> 
> If you are not familiar with Linux, then it is not a good approach. My next one would be trying your Pico on a friend's PC and see with Device Manager if it is recognized.

yeah ı will try that tomorrow if its because of my computer i will inform you guys

**Reply by UmutDemirel06** [reply]
yeah ı will try that tomorrow if it's because of my computer i will inform you guys

**Comment by beetlegigg** [comment]
Yes, the bit you show on your screen dump is what is on your Thonny computer, you need to scroll down to see what, if anything, in on the pico.  It will be below the thick bar on the bottom of your dump

**Reply by UmutDemirel06** [reply]
there seems to be nothing

**Comment by beetlegigg** [comment]
And just to check that you have hit Thonny's big red stop button a few times.  When you first connect your pico make sure you are seeing the REPL and its showing the version of micropython you are using, and on program failures the stop/restart button may need to be hit.

**Reply by UmutDemirel06** [reply]
that ı wanted ask as well repl menu isnt there

**Reply by UmutDemirel06** [reply]
ı was able to run blink in Arduino ide but that's about it ı couldn't get output in any way so ı tried to use thonny but that didn't upload anything to pico

**Comment by beetlegigg** [comment]
I think that if you run Arduino C++ on your pico you zap the micropython which will probably mean you need to reload mp.   This was certainly the case when using the Arduino IDE on an adafruit feather board with mp installed.   Arduino and micropython do not mix well.  
Just in case something on you pico is upsetting things you could put the flash_nuke.uf2 onto the pico before yet again reloading the mp.uf2 file.  The nuke will mean any files you may have put on the pico will be zapped as well as the mp.   
Other that that you could try to see what happens if you load a previous version of Thonny to you PC, but Thonny 4.1 works well for me (though I dont use a PC, only mac and rpi4)   I cant think of anything else to suggest.  I've been putting  mp on my pico's (some 15 of the blighters) and updating them without many issues apart for some initial connecting issues I mentioned in my first post.   Maybe its time to pause, have a cuppa, carefully read the rpi micropython installation instructions from the rpi micropython software developers kit documentation, and trying again from scratch.  Sorry not to have better suggestions for you.

**Reply by UmutDemirel06** [reply]
no worries man ıve been fighting with this problem for about a week ı even bought a new pico but it didn't work as well
